src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -348,3 +347,145 @@ def load_profile_properties(self, profile_name: str) -> dict:
+            value (str): The value to be set for the property.
+            profile_name (str): The name of the profile to set the property in.
+            secure (bool): If True, the property will be stored securely. Default is None.
+        """
+        if self.profiles is None:
+            self.init_from_file()
+
+        # Checking whether the property should be stored securely or in plain text
+        property_name = json_path.split(".")[-1]
+        profile_name = self.get_profile_name_from_path(json_path)
+        # check if the property is already secure
+        is_property_secure = self.__is_secure(profile_name, property_name)
+        is_secure = secure if secure is not None else is_property_secure
+
+        current_profile = self.find_profile(profile_name, self.profiles) or {}
+        current_properties = current_profile.setdefault("properties", {})
+        current_secure = current_profile.setdefault("secure", [])
+
+        if is_secure:
+            if not is_property_secure:
+                current_secure.append(property_name)
+
+            CredentialManager.secure_props[self.filepath] = {
+                **CredentialManager.secure_props.get(self.filepath, {}), json_path: value}
+            current_properties.pop(property_name, None)
+
+        else:
+            if is_property_secure:
+                current_secure.remove(property_name)
+            current_properties[property_name] = value
+
+        current_profile["properties"] = current_properties
+        current_profile["secure"] = current_secure
+        self.__set_or_create_nested_profile(profile_name, current_profile)
+
+    def set_profile(self, profile_path: str, profile_data: dict) -> None: 
+        """
+        Set a profile in the config file.
+
+        Parameters:
+            profile_path (str): The path of the profile to be set. eg: profiles.zosmf
+            profile_data (dict): The data to be set for the profile.
+        """
+        if self.profiles is None:
+            self.init_from_file()
+        profile_name = self.get_profile_name_from_path(profile_path)
+        if "secure" in profile_data:
+            # Checking if the profile has a 'secure' field with values
+            secure_fields = profile_data["secure"]
+            current_profile  = self.find_profile(profile_name,self.profiles) or {}
+            existing_secure_fields = current_profile.get("secure", [])
+            new_secure_fields = [field for field in secure_fields if field not in existing_secure_fields]
+
+            # JSON paths for new secure properties and store their values in CredentialManager.secure_props
+            for field in new_secure_fields:
+                json_path = f"{profile_path}.properties.{field}.value"
+                CredentialManager.secure_props[self.filepath] = {
+                    **CredentialManager.secure_props.get(self.filepath, {}),
+                    json_path: profile_data["properties"][field]
+                }
```

Comment:
When `set_profile` is called I think we want to overwrite secure props if they already exist for the profile.

Before the for loop, we could reset the object to be empty:
`CredentialManager.secure_props[self.filepath] = {}`
And inside the for loop, we could eliminate the spread operator:
`CredentialManager.secure_props[self.filepath][json_path] = profile_data["properties"][field]`